src/chocolatey/infrastructure.app/nuget/NugetList.cs

```diff
@@ -47,57 +47,81 @@ public static IEnumerable<IPackageSearchMetadata> GetPackages(ChocolateyConfigur
 
         public static int GetCount(ChocolateyConfiguration configuration, ILogger nugetLogger, IFileSystem filesystem)
         {
-            return execute_package_search(configuration, nugetLogger, filesystem).GetAwaiter().GetResult().Count();
+            var packageRepositories = NugetCommon.GetRemoteRepositories(configuration, nugetLogger, filesystem);
+            var packageRepositoriesResources = NugetCommon.GetRepositoryResources(packageRepositories);
+            string searchTermLower = configuration.Input.to_lower();
+
+            SearchFilter searchFilter = new SearchFilter(configuration.Prerelease);
+            searchFilter.IncludeDelisted = configuration.ListCommand.LocalOnly;
+
+
+            int totalCount = 0;
+            foreach (var repositoryResources in packageRepositoriesResources)
+            {
+                if (repositoryResources.searchResource == null)
+                {
+                    continue;
+                }
+
+                totalCount += repositoryResources.searchResource.SearchCountAsync(searchTermLower, searchFilter, nugetLogger, CancellationToken.None).GetAwaiter().GetResult();
+            }
+
+            return totalCount;
         }
 
         private async static Task<IQueryable<IPackageSearchMetadata>> execute_package_search(ChocolateyConfiguration configuration, ILogger nugetLogger, IFileSystem filesystem)
         {
             var packageRepositories = NugetCommon.GetRemoteRepositories(configuration, nugetLogger, filesystem);
             var packageRepositoriesResources = NugetCommon.GetRepositoryResources(packageRepositories);
             string searchTermLower = configuration.Input.to_lower();
+
             SearchFilter searchFilter = new SearchFilter(configuration.Prerelease);
             searchFilter.IncludeDelisted = configuration.ListCommand.LocalOnly;
-            var cacheContext = new ChocolateySourceCacheContext(configuration);
+            searchFilter.OrderBy = SearchOrderBy.Id;
 
+            if (configuration.ListCommand.OrderByPopularity)
+            {
+                searchFilter.OrderBy = SearchOrderBy.DownloadCount;

```

Comment:
Unless order by ID is specified on the command line, this should default to download count, unless I'm missing something. If order by popularity is specified on the command line, then at the end, a sort of the packages from all the repositories by download count should be done.

Reply:
> Unless order by ID is specified on the command line, this should default to download count, unless I'm missing something. 

Ordering by the identifier is the default in Chocolatey CLI, ordering by the popularity/download count is an opt-in feature when searching for packages through the CLI, so this code is currently correct.

> If order by popularity is specified on the command line, then at the end, a sort of the packages from all the repositories by download count should be done.

That is a good point, and I agree that should probably be done, both for the ID and the download count before we return all the results.

Reply:
Sounds good.